akka-javasdk/src/main/scala/akka/javasdk/impl/keyvalueentity/KeyValueEntityImpl.scala

```diff
@@ -205,5 +206,5 @@ private[impl] final class KeyValueEntityImpl[S, KV <: KeyValueEntity[S]](
     serializer.toBytes(obj)
 
   override def stateFromBytes(pb: BytesPayload): SpiEventSourcedEntity.State =
-    serializer.fromBytes(router.entityStateType, pb)
+    serializer.fromBytes(entityStateType, pb).asInstanceOf[SpiEventSourcedEntity.State]
```

Comment:
The type parameters are not very useful at this level, this cast existed already before, but then because cast of the router itself.